### PR TITLE
Migrate tests to Django's test harness

### PR DESCRIPTION
**Pull request by davish**
[ch2944]

PyTest is probably better than I'm giving it credit for, but I have a lot more experience in Django's test harness and so this PR takes advantage of async tests in Django 3.1 and moves all of the tests to that. Also factored out common behavior in tests so they're a lot shorter and a lot more readable.

The normal `setUp` and `tearDown` `TestCase` methods run in different threads from the actual test itself, and so wouldn't work for creating async objects like `WebsocketCommunicator`. I added a small decorator to the test file that will run the methods `asyncSetUp` and `asyncTearDown` at the right times. This decorator is exported in `rest_live.utils` for use in application-level test cases. 

Added a section in the docs for writing application-level tests.

A consequence is that we need to set a floor of Django 3.1, the newest version. Documentation was updated to reflect this.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [tox]
 envlist =
 ;    lint,
-    py36-django{30,31},
-    py37-django{30,31},
-    py38-django{30,31},
+    py36-django{31}
```
Comment (ArmaanT) [inline tox.ini:4]: Should we re-add testing django master + lint checking?

File: tox.ini [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+    py37-django{31}
+    py38-django{31}
 
 [testenv]
 commands = pytest --cov=rest_live --cov-append --junitxml=test-results/tests.xml
```
Comment (ArmaanT) [inline tox.ini:9]: This line will need to be changed to reflect testing with django/manage.py

**Review by ArmaanT — CHANGES_REQUESTED** [review]
Few comments. I do have one main concern though. Convention for packages like this is definitely to use pytest instead of Django for testing, Switching this package over would make it different from all of our other packages and probably more confusing to whoever is maintaining it in the future.

File: tox.ini (hunk 1 above)
Comment (davish) [reply tox.ini:4]: Async is so unstable in django master that I don't think it's a good bet that channels will play nicely with it -- I checked their CI and it doesn't look like they're testing against it.

File: setup.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def run(self):
 setup(
     name="django-rest-live",
     version=VERSION,
-    packages=find_packages(exclude=["tests", "test_app"]),
+    packages=find_packages(include=["rest_live"]),
```
[the remark this answers is not in the document]
Comment (davish) [reply setup.py:30]: Looks like `packages` doesn't define the whole bundle but just the importable python files: https://packaging.python.org/glossary/#term-Import-Package. I switched to `include` here to make sure we don't ship `manage.py`.

File: tox.ini (hunk 1 above)
Comment (davish) [reply tox.ini:4]: Will add back `lint`, though.

Comment (davish) [reply tox.ini:4]: `lint` erroring for no reason. Will write a ticket to look into it.

File: setup.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (davish) [reply setup.py]: Switched it back in either case

**Pull request merged**